src/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh

```diff
@@ -23,7 +23,8 @@ if [[ "$VSCODE_INJECTION" == "1" ]]; then
 		ZDOTDIR=$VSCODE_ZDOTDIR
 	fi
 
-	if [[ -f $USER_ZDOTDIR/.zsh_history && -z $HISTFILE ]]; then
+	if [[ -f $USER_ZDOTDIR/.zsh_history && -f $HISTFILE ]]; then
```

Comment:
this was never getting entered when there was a `$HISTFILE` because `-z` was used instead of `-f`. 

so `HISTFILE` was never getting set to be the user's